### ISSUE TITLE
add hex and base64 encoding

### ISSUE THREAD

**Issue by Ma124** [d5/tengo#216]
I'm currently working on a [crypto module](https://github.com/Ma124/tengo/tree/stdlib-crypto) #135 and often you want your encrypted data base64 encoded or your hashes as hex.

Which is why I suggest adding these features to the text module or an encoding module with support for base64 encoding and hex because the only way I known to hex encode bytes at the moment is `sprintf("%x", bin)` which is neither immediately obvious nor efficient.

With some featback from @d5 I can create a pull request.

**Comment by mccolljr** [comment]
@Ma124 I would prefer to keep encoding out of the `text` module since it is also applicable to non-text data. I think the optimal approach here is what Go does, where we have a separate package for each encoding scheme (base64, hex, etc) that provide a similar interface for encoding/decoding data. What are your thoughts on that approach?

**Comment by Ma124** [comment]
@mccolljr I'd prefer all encodings in one module so you only need to import it once, a module with just two functions `func EncodeHex(bytes)` and `func DecodeHex(string)` would kind of destroy the point of modules and clutter the stdlib with countless modules if we then decide to not only to include hex and base64 but also for example base32 or ascii85 you would need this call:
```go
modules := objects.NewModuleMap()

modules.AddBuiltinModule("hex", BuiltinModules["hex"])
modules.AddBuiltinModule("base64", BuiltinModules["base64"])
modules.AddBuiltinModule("base32", BuiltinModules["base32"])
modules.AddBuiltinModule("ascii85", BuiltinModules["ascii85"])

s.SetImports(modules)
```
instead of a simple:
```go
s.SetImports(stdlib.GetModuleMap("encoding"))
```

and all the encodings are linked with the built executable anyway.

**Comment by geseq** [comment]
I'm in agreement with @mccolljr. We already have a `json` package, which was fine as a separate package while it was the only one available, but as we're considering adding more here I'd rather avoid importing all functions into the runtime in one go. I'd suggest moving `json` into encoding and adding each scheme into a separate package.

The upside is that it also keeps it in line with Go.

**Comment by mccolljr** [comment]
Here's my concern. If we add all the functions to a single encoding module, we're going to have a package with a huge surface area and as a result the function names are going to need to be specific and long to compensate. We can define a meta-module that re-exports the different encoding schemes under one module as functions, but I believe there is value in having each encoding module set up separately. You can then use an encoder/decoder as a defacto interface, and pass around things with the "encode" and "decode" functions as properties.

**Comment by Ma124** [comment]
Ok I'll submit a pull request with a module for base64 and one for hex.

**Comment by Ma124** [comment]
Fixed by #221
